rotation.go

```diff
@@ -256,21 +257,21 @@ func (b *backend) findStaticWAL(ctx context.Context, s logical.Storage, id strin
 		return nil, nil
 	}
 
-	data := wal.Data.(map[string]interface{})
 	walEntry := setCredentialsWAL{
-		walID:          id,
-		walCreatedAt:   wal.CreatedAt,
-		NewPassword:    data["new_password"].(string),
-		RoleName:       data["role_name"].(string),
-		Username:       data["username"].(string),
-		DN:             data["dn"].(string),
-		PasswordPolicy: data["password_policy"].(string),
-	}
-	lvr, err := time.Parse(time.RFC3339, data["last_vault_rotation"].(string))
+		walID:        id,
+		walCreatedAt: wal.CreatedAt,
+	}
+	d, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		DecodeHook: mapstructure.StringToTimeHookFunc(time.RFC3339),
```

Comment:
This is needed because the missing entry you mentioned is the PasswordPolicy, which was added in the previous bugfix right?

Also, I haven't seen a custom DecoderConfig used like this until now. Am I interpreting the API docs for DecodeHook correctly - decode will be applied to every value in the struct, but `StringToTimeHookFunc` checks if a string is a parseable `time.Time` value and does nothing if it isn't but converts the time if it is?

Reply:
Yep, that's right! I followed the same pattern in the code which was not checking the WAL data (a map) for existence of each key (see [rotation.go#L263-L267](https://github.com/hashicorp/vault-plugin-secrets-openldap/blob/main/rotation.go#L263-L267)). This is what caused a panic when "password_policy" was empty.

That's correct. If the source field is a string type and the target field is a time.Time type, `StringToTimeHookFunc` will convert the string to a time.Time with the given format (`time.RFC3339`).